test/unit/api/state.js

```diff
@@ -1,48 +1,63 @@
+import afterMutations from '../../lib/after-mutations';
 import element from '../../lib/element';
+import fixture from '../../lib/fixture';
 import { state } from '../../../src/index';
 
 describe('api/state', function () {
-  function create () {
-    return new (element().skate({
+  let elem;
+
+  beforeEach(done => {
+    elem = new (element().skate({
       props: {
         prop1: {
           initial: 'test1'
         },
         prop2: {
           initial: 'test2'
-        },
-        rendered: {
-          default: 0
         }
       },
-      render (elem) {
-        elem.rendered++;
+      created(elem) {
+        elem._rendered = 0;
+      },
+      render(elem) {
+        elem._rendered++;
       }
     }));
-  }
-
-  it('should return only defined properties', function () {
-    const curr = state(create());
-    expect(curr.prop1).to.equal('test1');
-    expect(curr.prop2).to.equal('test2');
+    fixture(elem);
+    afterMutations(done);
   });
 
-  it('should set all properties', function () {
-    const elem = create();
-    state(elem, {
-      prop1: 'updated1',
-      prop2: 'updated2',
-      undeclaredProp: 'updated3'
+  it('getting', () => {
+    it('should return only defined properties', function () {
+      const curr = state(elem);
+      expect(curr.prop1).to.equal('test1');
+      expect(curr.prop2).to.equal('test2');
+      expect(curr.undeclaredProp).to.equal(undefined);
```

Comment:
I reorganised the tests here. The red was for setting, the green is for getting. Setting sets any prop. Getting only returns `props` you've defined for the component.